internal/storage/sbom_store.go

```diff
@@ -0,0 +1,525 @@
+//nolint:wrapcheck // We want to return the errors from k8s.io/apiserver/pkg/storage as they are.
+package storage
+
+import (
+	"context"
+	"database/sql"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"log"
+	"strings"
+
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/watch"
+	"k8s.io/apiserver/pkg/registry/generic"
+	genericregistry "k8s.io/apiserver/pkg/registry/generic/registry"
+	"k8s.io/apiserver/pkg/registry/rest"
+	"k8s.io/apiserver/pkg/storage"
+
+	sq "github.com/Masterminds/squirrel"
+	"github.com/jmoiron/sqlx"
+
+	"github.com/rancher/sbombastic/api/storage/v1alpha1"
+)
+
+// NewSBOMStore returns a store registry that will work against API services.
+func NewSBOMStore(scheme *runtime.Scheme, optsGetter generic.RESTOptionsGetter, db *sqlx.DB) (*genericregistry.Store, error) {
+	strategy := NewStrategy(scheme)
+
+	store := &genericregistry.Store{
+		NewFunc:                   func() runtime.Object { return &v1alpha1.SBOM{} },
+		NewListFunc:               func() runtime.Object { return &v1alpha1.SBOMList{} },
+		PredicateFunc:             MatchSBOM,
+		DefaultQualifiedResource:  v1alpha1.Resource("sboms"),
+		SingularQualifiedResource: v1alpha1.Resource("sbom"),
+		Storage: genericregistry.DryRunnableStorage{
+			Storage: &sbomStore{
+				broadcaster: watch.NewBroadcaster(1000, watch.WaitIfChannelFull),
+				db:          db,
+			},
+		},
+		CreateStrategy: strategy,
+		UpdateStrategy: strategy,
+		DeleteStrategy: strategy,
+
+		// TODO: define table converter that exposes more than name/creation timestamp
+		TableConvertor: rest.NewDefaultTableConvertor(v1alpha1.Resource("sboms")),
+	}
+
+	options := &generic.StoreOptions{RESTOptions: optsGetter, AttrFunc: GetAttrs}
+	if err := store.CompleteWithOptions(options); err != nil {
+		return nil, fmt.Errorf("unable to complete store with options: %w", err)
+	}
+
+	return store, nil
+}
+
+var _ storage.Interface = &sbomStore{}
```

Comment:
What is the purpose of that?

Reply:
This convention ensures that the code will not compile if we do not implement the `storage.Interface` correctly.